File: .sdk-repos/orchestration-cluster-api-csharp/examples/BatchOperation.cs
// Compilable usage examples for batch operation management.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class BatchOperationExamples
{
    #region GetBatchOperation
    // <GetBatchOperation>
    public static async Task GetBatchOperationExample(BatchOperationKey batchOperationKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetBatchOperationAsync(
            batchOperationKey);

        Console.WriteLine($"Batch operation: {result.BatchOperationKey}");
    }
    // </GetBatchOperation>
    #endregion GetBatchOperation

    #region SearchBatchOperations

    // <SearchBatchOperations>
    public static async Task SearchBatchOperationsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchBatchOperationsAsync(
            new BatchOperationSearchQuery());

        foreach (var op in result.Items)
        {
            Console.WriteLine($"Batch operation: {op.BatchOperationKey}");
        }
    }
    // </SearchBatchOperations>
    #endregion SearchBatchOperations

    #region SearchBatchOperationItems

    // <SearchBatchOperationItems>
    public static async Task SearchBatchOperationItemsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchBatchOperationItemsAsync(
            new BatchOperationItemSearchQuery());

        foreach (var item in result.Items)
        {
            Console.WriteLine($"Item: {item.ItemKey}");
        }
    }
    // </SearchBatchOperationItems>
    #endregion SearchBatchOperationItems

    #region CancelBatchOperation

    // <CancelBatchOperation>
    public static async Task CancelBatchOperationExample(BatchOperationKey batchOperationKey)
    {
        using var client = CamundaClient.Create();

        await client.CancelBatchOperationAsync(batchOperationKey);
    }
    // </CancelBatchOperation>
    #endregion CancelBatchOperation

    #region SuspendBatchOperation

    // <SuspendBatchOperation>
    public static async Task SuspendBatchOperationExample(BatchOperationKey batchOperationKey)
    {
        using var client = CamundaClient.Create();

        await client.SuspendBatchOperationAsync(batchOperationKey);
    }
    // </SuspendBatchOperation>
    #endregion SuspendBatchOperation

    #region ResumeBatchOperation

    // <ResumeBatchOperation>
    public static async Task ResumeBatchOperationExample(BatchOperationKey batchOperationKey)
    {
        using var client = CamundaClient.Create();

        await client.ResumeBatchOperationAsync(batchOperationKey);
    }
    // </ResumeBatchOperation>
    #endregion ResumeBatchOperation

    #region CancelProcessInstancesBatchOperation

    // <CancelProcessInstancesBatchOperation>
    public static async Task CancelProcessInstancesBatchOperationExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.CancelProcessInstancesBatchOperationAsync(
            new ProcessInstanceCancellationBatchOperationRequest());

        Console.WriteLine($"Batch operation key: {result.BatchOperationKey}");
    }
    // </CancelProcessInstancesBatchOperation>
    #endregion CancelProcessInstancesBatchOperation

    #region DeleteProcessInstancesBatchOperation

    // <DeleteProcessInstancesBatchOperation>
    public static async Task DeleteProcessInstancesBatchOperationExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.DeleteProcessInstancesBatchOperationAsync(
            new ProcessInstanceDeletionBatchOperationRequest());

        Console.WriteLine($"Batch operation key: {result.BatchOperationKey}");
    }
    // </DeleteProcessInstancesBatchOperation>
    #endregion DeleteProcessInstancesBatchOperation

    #region MigrateProcessInstancesBatchOperation

    // <MigrateProcessInstancesBatchOperation>
    public static async Task MigrateProcessInstancesBatchOperationExample(ProcessDefinitionKey targetProcessDefinitionKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.MigrateProcessInstancesBatchOperationAsync(
            new ProcessInstanceMigrationBatchOperationRequest
            {
                Filter = new ProcessInstanceFilter(),
                MigrationPlan = new ProcessInstanceMigrationBatchOperationPlan
                {
                    TargetProcessDefinitionKey = targetProcessDefinitionKey,
                },
            });

        Console.WriteLine($"Batch operation key: {result.BatchOperationKey}");
    }
    // </MigrateProcessInstancesBatchOperation>
    #endregion MigrateProcessInstancesBatchOperation

    #region ModifyProcessInstancesBatchOperation

    // <ModifyProcessInstancesBatchOperation>
    public static async Task ModifyProcessInstancesBatchOperationExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.ModifyProcessInstancesBatchOperationAsync(
            new ProcessInstanceModificationBatchOperationRequest());

        Console.WriteLine($"Batch operation key: {result.BatchOperationKey}");
    }
    // </ModifyProcessInstancesBatchOperation>
    #endregion ModifyProcessInstancesBatchOperation

    #region ResolveIncidentsBatchOperation

    // <ResolveIncidentsBatchOperation>
    public static async Task ResolveIncidentsBatchOperationExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.ResolveIncidentsBatchOperationAsync(
            new ProcessInstanceIncidentResolutionBatchOperationRequest());

        Console.WriteLine($"Batch operation key: {result.BatchOperationKey}");
    }
    // </ResolveIncidentsBatchOperation>
    #endregion ResolveIncidentsBatchOperation

    #region DeleteDecisionInstancesBatchOperation

    // <DeleteDecisionInstancesBatchOperation>
    public static async Task DeleteDecisionInstancesBatchOperationExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.DeleteDecisionInstancesBatchOperationAsync(
            new DecisionInstanceDeletionBatchOperationRequest());

        Console.WriteLine($"Batch operation key: {result.BatchOperationKey}");
    }
    // </DeleteDecisionInstancesBatchOperation>
    #endregion DeleteDecisionInstancesBatchOperation
}

File: .sdk-repos/version-8.9/orchestration-cluster-api-python/examples/batch_operation.py
# Compilable usage examples for batch operation management.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    BatchOperationItemSearchQuery,
    BatchOperationKey,
    BatchOperationSearchQuery,
    CamundaClient,
    DecisionInstanceDeletionBatchOperationRequest,
    DecisionInstanceDeletionBatchOperationRequestFilter,
    ElementId,
    MigrateProcessInstanceMappingInstruction,
    ProcessDefinitionKey,
    ProcessInstanceCancellationBatchOperationRequest,
    ProcessInstanceCancellationBatchOperationRequestFilter,
    ProcessInstanceDeletionBatchOperationRequest,
    ProcessInstanceIncidentResolutionBatchOperationRequest,
    ProcessInstanceMigrationBatchOperationRequest,
    ProcessInstanceMigrationBatchOperationRequestMigrationPlan,
    ProcessInstanceModificationBatchOperationRequest,
    ProcessInstanceModificationMoveBatchOperationInstruction,
    Unset,
)


# region GetBatchOperation
def get_batch_operation_example(batch_operation_key: BatchOperationKey) -> None:
    client = CamundaClient()

    result = client.get_batch_operation(
        batch_operation_key=batch_operation_key,
    )

    print(f"Batch operation: {result.batch_operation_key}")
# endregion GetBatchOperation


# region SearchBatchOperations
def search_batch_operations_example() -> None:
    client = CamundaClient()

    result = client.search_batch_operations(
        data=BatchOperationSearchQuery(),
    )

    if not isinstance(result.items, Unset):
        for op in result.items:
            print(f"Batch operation: {op.batch_operation_key}")
# endregion SearchBatchOperations


# region SearchBatchOperationItems
def search_batch_operation_items_example(batch_operation_key: BatchOperationKey) -> None:
    client = CamundaClient()

    result = client.search_batch_operation_items(
        batch_operation_key=batch_operation_key,
        data=BatchOperationItemSearchQuery(),
    )

    if not isinstance(result.items, Unset):
        for item in result.items:
            print(f"Item: {item.item_key}")
# endregion SearchBatchOperationItems


# region CancelBatchOperation
def cancel_batch_operation_example(batch_operation_key: BatchOperationKey) -> None:
    client = CamundaClient()

    client.cancel_batch_operation(
        batch_operation_key=batch_operation_key,
    )
# endregion CancelBatchOperation


# region SuspendBatchOperation
def suspend_batch_operation_example(batch_operation_key: BatchOperationKey) -> None:
    client = CamundaClient()

    client.suspend_batch_operation(
        batch_operation_key=batch_operation_key,
    )
# endregion SuspendBatchOperation


# region ResumeBatchOperation
def resume_batch_operation_example(batch_operation_key: BatchOperationKey) -> None:
    client = CamundaClient()

    client.resume_batch_operation(
        batch_operation_key=batch_operation_key,
    )
# endregion ResumeBatchOperation


# region CancelProcessInstancesBatchOperation
def cancel_process_instances_batch_operation_example() -> None:
    client = CamundaClient()

    result = client.cancel_process_instances_batch_operation(
        data=ProcessInstanceCancellationBatchOperationRequest(
            filter_=ProcessInstanceCancellationBatchOperationRequestFilter(),
        ),
    )

    print(f"Batch operation key: {result.batch_operation_key}")
# endregion CancelProcessInstancesBatchOperation


# region DeleteProcessInstancesBatchOperation
def delete_process_instances_batch_operation_example() -> None:
    client = CamundaClient()

    result = client.delete_process_instances_batch_operation(
        data=ProcessInstanceDeletionBatchOperationRequest(
            filter_=ProcessInstanceCancellationBatchOperationRequestFilter(),
        ),
    )

    print(f"Batch operation key: {result.batch_operation_key}")
# endregion DeleteProcessInstancesBatchOperation


# region MigrateProcessInstancesBatchOperation
def migrate_process_instances_batch_operation_example(target_process_definition_key: ProcessDefinitionKey, source_element_id: ElementId, target_element_id: ElementId) -> None:
    client = CamundaClient()

    result = client.migrate_process_instances_batch_operation(
        data=ProcessInstanceMigrationBatchOperationRequest(
            filter_=ProcessInstanceCancellationBatchOperationRequestFilter(),
            migration_plan=ProcessInstanceMigrationBatchOperationRequestMigrationPlan(
                target_process_definition_key=target_process_definition_key,
                mapping_instructions=[
                    MigrateProcessInstanceMappingInstruction(
                        source_element_id=source_element_id,
                        target_element_id=target_element_id,
                    ),
                ],
            ),
        ),
    )

    print(f"Batch operation key: {result.batch_operation_key}")
# endregion MigrateProcessInstancesBatchOperation


# region ModifyProcessInstancesBatchOperation
def modify_process_instances_batch_operation_example(source_element_id: ElementId, target_element_id: ElementId) -> None:
    client = CamundaClient()

    result = client.modify_process_instances_batch_operation(
        data=ProcessInstanceModificationBatchOperationRequest(
            filter_=ProcessInstanceCancellationBatchOperationRequestFilter(),
            move_instructions=[
                ProcessInstanceModificationMoveBatchOperationInstruction(
                    source_element_id=source_element_id,
                    target_element_id=target_element_id,
                ),
            ],
        ),
    )

    print(f"Batch operation key: {result.batch_operation_key}")
# endregion ModifyProcessInstancesBatchOperation


# region ResolveIncidentsBatchOperation
def resolve_incidents_batch_operation_example() -> None:
    client = CamundaClient()

    result = client.resolve_incidents_batch_operation(
        data=ProcessInstanceIncidentResolutionBatchOperationRequest(
            filter_=ProcessInstanceCancellationBatchOperationRequestFilter(),
        ),
    )

    print(f"Batch operation key: {result.batch_operation_key}")
# endregion ResolveIncidentsBatchOperation


# region DeleteDecisionInstancesBatchOperation
def delete_decision_instances_batch_operation_example() -> None:
    client = CamundaClient()

    result = client.delete_decision_instances_batch_operation(
        data=DecisionInstanceDeletionBatchOperationRequest(
            filter_=DecisionInstanceDeletionBatchOperationRequestFilter(),
        ),
    )

    print(f"Batch operation key: {result.batch_operation_key}")
# endregion DeleteDecisionInstancesBatchOperation
